### PR TITLE
🎨 Utilizing tolerance to mitigate floating-point inaccuracies

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

This PR addresses the floating-point errors that caused the failure on macOS for #390 by introducing a tolerance in the energy distribution calculations. Additionally, it enhances user-friendliness by implementing an enum to control the updating of the charge index when changing the charge state.

## Checklist:

- [ ] The pull request only contains commits that are related to it.
- [ ] I have added appropriate tests and documentation.
- [ ] I have added a changelog entry.
- [x] I have created/adjusted the Python bindings for any new or updated functionality.
- [x] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: bindings/pyfiction/include/pyfiction/technology/charge_distribution_surface.hpp [hunk 1]
```diff
@@ -294,6 +293,17 @@ inline void charge_distribution_surfaces(pybind11::module& m)
...
+    /**
+     * Assign charge state mode.
+     */
+    py::enum_<fiction::charge_index_mode>(m, "charge_index_mode", DOC(fiction_energy_calculation), py::module_local())
+        .value("KEEP_OLD_ENERGY_VALUE", fiction::charge_index_mode::UPDATE_CHARGE_INDEX,
+               DOC(fiction_charge_index_mode_UPDATE_CHARGE_INDEX))
+        .value("UPDATE_ENERGY", fiction::charge_index_mode::DO_NOT_UPDATE_CHARGE_INDEX,
+               DOC(fiction_charge_index_mode_DO_NOT_UPDATE_CHARGE_INDEX))
+
+        ;
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/technology/charge_distribution_surface.hpp:305]: Could you add a Python test involving this `enum`?

Comment (Drewniok) [reply bindings/pyfiction/include/pyfiction/technology/charge_distribution_surface.hpp:305]: sure, good point!

File: bindings/pyfiction/test/technology/test_charge_distribution_surface.py [hunk 2]
```diff
@@ -50,18 +50,27 @@ def test_initialization_111_lattice(self):
...
         self.assertEqual(charge_lyt.get_charge_state((2, 1)), sidb_charge_state.NEGATIVE)
 
-        charge_lyt.assign_charge_state((0, 1), sidb_charge_state.NEUTRAL)
-        charge_lyt.assign_charge_state((1, 1), sidb_charge_state.NEGATIVE)
-        charge_lyt.assign_charge_state((2, 1), sidb_charge_state.NEUTRAL)
+        self.assertEqual(charge_lyt.get_charge_index_and_base()[0], 0)
+
+        charge_lyt.assign_charge_state((0, 1), sidb_charge_state.NEUTRAL, charge_index_mode.DO_NOT_UPDATE_CHARGE_INDEX)
+        charge_lyt.assign_charge_state((1, 1), sidb_charge_state.NEGATIVE, charge_index_mode.DO_NOT_UPDATE_CHARGE_INDEX)
+        charge_lyt.assign_charge_state((2, 1), sidb_charge_state.NEUTRAL, charge_index_mode.DO_NOT_UPDATE_CHARGE_INDEX)
```
Comment (marcelwa) [inline bindings/pyfiction/test/technology/test_charge_distribution_surface.py:57]: The naming ran out of sync here

File: bindings/pyfiction/include/pyfiction/technology/charge_distribution_surface.hpp [hunk 3]
```diff
@@ -294,6 +293,17 @@ inline void charge_distribution_surfaces(pybind11::module& m)
 
         ;
 
+    /**
+     * Assign charge state mode.
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/technology/charge_distribution_surface.hpp:297]: This comment seems to be inconsistent given the others in this file.

**Pull request merged**